### ISSUE TITLE
Callback for NSAttributedString click event 

### ISSUE THREAD

**Issue by GeorgeNance** [kyle-n/HighlightedTextView#11]
**Is your feature request related to a problem? Please describe.**
Yes, there is no clear way to handle tap events right now when creating "links"

**Describe the solution you'd like**
A callback similar to the existing ones when a link is clicked.

**Describe alternatives you've considered**
None yet

**Additional context**
https://stackoverflow.com/a/50767617

**Comment by kyle-n** [comment]
You can _almost_ do this in the existing code. 

```swift
fileprivate let urlRegexPattern = "((?:http|https)://)?(?:www\\.)?[\\w\\d\\-_]+\\.\\w{2,3}(\\.\\w{2})?(/(?<=/)(?:[\\w\\d\\-./_]+)?)?"
fileprivate let urlRegex = try! NSRegularExpression(pattern: urlRegexPattern, options: [])

struct ContentView: View {
    @State private var text: String = ""
    @State private var currentEditor: EditorType = .markdown
    
    var body: some View {
        HighlightedTextEditor(text: $text, highlightRules: [
            HighlightRule(pattern: urlRegex, formattingRules: [
                TextFormattingRule(key: .underlineStyle, value: NSUnderlineStyle.single.rawValue),
                TextFormattingRule(key: .foregroundColor, value: UIColor.blue),
                TextFormattingRule(key: .link, value: "https://www.google.com")
            ])
        ])
    }
}
```

But, of course, then every link goes to the Google homepage. 

Let me think about how to do this. The information needed to make a tappable link is all there in `HighlightingTextEditor`, we just have to expose it in a clean way. 

Do you want arbitrary callbacks or just link functionality on tap of a highlighted section?

**Comment by GeorgeNance** [comment]
Hmm. Arbitrary callbacks is what I am leaning toward. But only if its possible to identify if what rule was clicked on.

**Comment by kyle-n** [comment]
Hello, it has been quite a while. This feature was delayed by work on #25 and #32, but it has shipped in `1.6.0`. I could not figure out how to make arbitrary callbacks work, but you can do tappable links now. See the [release notes](https://github.com/kyle-n/HighlightedTextEditor/releases/tag/1.6.0) for more.
